native/cocos/audio/apple/AudioCache.mm

```diff
@@ -23,284 +23,229 @@ of this software and associated engine source code (the "Software"), a limited,
+//                                            sampleRate:(double)44100
+//                                            channels:2
+//                                            interleaved:true];
+AudioDataFormat formatConverter(AVAudioCommonFormat originalAppleFormat){
+    switch (originalAppleFormat) {
+        case AVAudioPCMFormatInt16:
+            return AudioDataFormat::SIGNED_16;
+        case AVAudioPCMFormatInt32:
+            return AudioDataFormat::SIGNED_32;
+        case AVAudioPCMFormatFloat32:
+            return AudioDataFormat::FLOAT_32;
+        case AVAudioPCMFormatFloat64:
+            return AudioDataFormat::FLOAT_64;
+        default:
+            return AudioDataFormat::UNKNOWN;
+    }
 }
-
-@end
-
-@implementation NSTimerWrapper
-
-- (id)initWithTimeInterval:(double)seconds callback:(const std::function<void()> &)cb {
-    if (self = [super init]) {
-        _timeoutCallback = cb;
-        NSTimer *timer = [NSTimer timerWithTimeInterval:seconds target:self selector:@selector(onTimeoutCallback:) userInfo:nil repeats:NO];
-        [[NSRunLoop currentRunLoop] addTimer:timer forMode:NSDefaultRunLoopMode];
+// return byte length of format correspond.
+uint32_t formatConvertToLength(AVAudioCommonFormat originalAppleFormat){
+    switch (originalAppleFormat) {
+        case AVAudioPCMFormatInt16:
+            return 2;
+        case AVAudioPCMFormatInt32:
+        case AVAudioPCMFormatFloat32:
+            return 4;
+        case AVAudioPCMFormatFloat64:
+            return 8;
+        default:
+            return 0;
     }
-
-    return self;
 }
-
-- (void)onTimeoutCallback:(NSTimer *)timer {
-    if (_timeoutCallback != nullptr) {
-        _timeoutCallback();
-        _timeoutCallback = nullptr;
+namespace cc {
+
+AudioCache::AudioCache(std::string &fileFullPath) {
+    _descriptor = {nil, nil};
+    NSString * path = [[NSString alloc] initWithUTF8String:fileFullPath.c_str()];
+    NSURL * url = [[NSURL alloc] initWithString:path];
+    NSError * err;
+    _descriptor.audioFile = [[AVAudioFile alloc] initForReading:url error:&err];
+    if (_descriptor.audioFile == nil) {
+        NSLog(@"AVAudioFile Error: ", [err localizedDescription]);
+        [err release];
+        assert([err retainCount] == 0);
+    }
+    [url release];
+    [path release];
+    loadState = State::READY;
+    _pcmHeader = {
+        static_cast<uint32_t>(_descriptor.audioFile.length),
+        _descriptor.audioFile.fileFormat.channelCount * formatConvertToLength( _descriptor.audioFile.fileFormat.commonFormat),
+        static_cast<uint32_t>(_descriptor.audioFile.fileFormat.sampleRate),
+        _descriptor.audioFile.fileFormat.channelCount,
+        formatConverter( _descriptor.audioFile.fileFormat.commonFormat)
+    };
+    _fileFullPath = fileFullPath;
+    // When audio length is bigger than MAX_BUFFER_LENGTH, should make it as a streaming audio.
+    if (_pcmHeader.totalFrames * _pcmHeader.bytesPerFrame > MAX_BUFFER_LENGTH) {
```

Comment:
I think is it better to use `MAX_FRAME_COUNT` to decide `_isStreaming`. `MAX_BUFFER_LENGTH` need to set carefully, it may break one frame data into two part.